Adyen/client.py

```diff
@@ -449,86 +358,42 @@ def call_checkout_api(self, request_data, method, endpoint, idempotency_key=None
-
-        return adyen_result
-
-    def call_management_api(self, request_data, method, endpoint, idempotency_key=None,
-                            **kwargs):
-        """This will call the management adyen api. xapi key merchant_account,
-        and platform are pulled from root module level and or self object.
-        AdyenResult will be returned on 200 response. Otherwise, an exception
-        is raised.
-
-        Args:
-            idempotency_key: https://docs.adyen.com/development-resources
-            /api-idempotency
-            request_data (dict): The dictionary of the request to place. This
-                should be in the structure of the Adyen API.
-                https://docs.adyen.com/api-explorer/#/CheckoutService
-            method (str): This is the method used to send the request to an endpoint.
-            endpoint (str): The specific endpoint of the API service to be called
-        """
-        if not self.http_init:
-            self._init_http_client()
-
-        # xapi at self object has highest priority. fallback to root module
-        # and ensure that it is set.
-        xapikey = False
-        if self.xapikey:
-            xapikey = self.xapikey
-        elif 'xapikey' in kwargs:
-            xapikey = kwargs.pop("xapikey")
 
-        if not xapikey:
-            errorstring = """Please set your webservice xapikey.
-             You can do this by running 'Adyen.xapikey = 'Your xapikey'"""
-            raise AdyenInvalidRequestError(errorstring)
-
-        # platform at self object has highest priority. fallback to root module
-        # and ensure that it is set to either 'live' or 'test'.
-        platform = None
-        if self.platform:
-            platform = self.platform
-        elif 'platform' in kwargs:
-            platform = kwargs.pop('platform')
-
-        if not isinstance(platform, str):
-            errorstring = "'platform' value must be type of string"
-            raise TypeError(errorstring)
-        elif platform.lower() not in ['live', 'test']:
-            errorstring = "'platform' must be the value of 'live' or 'test'"
-            raise ValueError(errorstring)
-
-        # Adyen requires this header to be set and uses the combination of
-        # merchant account and merchant reference to determine uniqueness.
-        headers = {}
-        if idempotency_key:
-            headers[self.IDEMPOTENCY_HEADER_NAME] = idempotency_key
-        url = self._determine_management_url(platform, endpoint)
+        url = self._determine_api_url(platform, service, endpoint)
+        # Match urls that require a live prefix and do not have one
+        if match("https://None-.*", url):
+            errorstring = """Please set your live suffix. You can set it
+                               by running 'settings.
+                               ENDPOINT_CHECKOUT_LIVE_SUFFIX = 'Your live suffix'"""
+            raise AdyenEndpointInvalidFormat(errorstring)
```

Comment:
Let's move this check to `_determine_api_url`. We can also implement in bit more explicit way by defining `PAL_LIVE_ENDPOINT_URL_TEMPLATE` (`settings.py`) as: 

```
PAL_LIVE_ENDPOINT_URL_TEMPLATE = "https://{live_prefix}-pal-live" \
                                 ".adyenpayments.com/pal/servlet"
```

Then checking it with:

```suggestion
        # Match urls that require a live prefix and do not have one
        if platform is 'live' and '{live_prefix}' in url:
            errorstring = """Please set your live suffix. You can set it
                               by running 'settings.
                               ENDPOINT_CHECKOUT_LIVE_SUFFIX = 'Your live suffix'"""
            raise AdyenEndpointInvalidFormat(errorstring)
```

It's good to document this requirement in `README.md`, so that people can be aware of it before going live.